src/openassetio-python/cmodule/src/errors/exceptionsBinding.cpp

```diff
@@ -74,3 +76,71 @@ void registerErrors(const py::module &mod) {
     }
   });
 }
+
+void registerBatchElementException(const py::module &mod) {
+  // Pybind has very limited support for custom exception types. This is
+  // a well-known tricky issue and is apparently not on the roadmap to
+  // fix. The only direct support is for an exception type that takes a
+  // single string parameter (message). However, we need `index` and
+  // `error` parameters to mirror C++.
+  //
+  // A way forward is provided by the sketch in:
+  // https://github.com/pybind/pybind11/issues/1281#issuecomment-1375950333
+  //
+  // We execute a Python string literal to create our exception
+  // type. The `globals` and `locals` dict parameters dictate the scope
+  // of execution, so we use this to ensure the definition is scoped to
+  // our `_openassetio` module.
+
+  py::exec(R"pybind(
+class BatchElementException(RuntimeError):
```

Comment:
[fix] Inherit from `OpenAssetIOException`.  

Might be worth reviving some of the reverted tests of exception hierarchy.

Reply:
Can't believe i made this mistake _twice_.

https://github.com/OpenAssetIO/OpenAssetIO/pull/1109/commits/84f9425d425e891cbd8985483fe0564d893fb0cf

I've added just a simple test for this, i'd hope we wouldn't need anything more than that for the single type.